applications/clock_app/clock_app.c

```diff
@@ -0,0 +1,412 @@
+            }
+        } else if(songSelect == 2) {
+            if(timerSecs == 80) {
+                NotificationApp* notification = furi_record_open("notification");
+                notification_message(notification, &clock_alert_mario1);
+                furi_record_close("notification");
+            }
+            if(timerSecs == 81) {
+                NotificationApp* notification = furi_record_open("notification");
+                notification_message(notification, &clock_alert_mario2);
+                furi_record_close("notification");
+            }
+            if(timerSecs == 82) {
+                NotificationApp* notification = furi_record_open("notification");
+                notification_message(notification, &clock_alert_mario3);
+                furi_record_close("notification");
+            }
+        } else {
+            if(timerSecs == 80) {
+                NotificationApp* notification = furi_record_open("notification");
+                notification_message(notification, &clock_alert_silent);
+                furi_record_close("notification");
+            }
+        }
+    }
+    // It's OK to loose this event if system overloaded
+    furi_message_queue_put(event_queue, &event, 0);
+}
+
+int32_t clock_app(void* p) {
+    UNUSED(p);
+    timerStarted = false;
+    timerSecs = 0;
+    songSelect = 2;
+    FuriMessageQueue* event_queue = furi_message_queue_alloc(8, sizeof(PluginEvent));
+    ClockState* plugin_state = malloc(sizeof(ClockState));
+    clock_state_init(plugin_state);
+    ValueMutex state_mutex;
+    if(!init_mutex(&state_mutex, plugin_state, sizeof(ClockState))) {
+        FURI_LOG_E(TAG, "cannot create mutex\r\n");
+        furi_message_queue_free(event_queue);
+        free(plugin_state);
+        return 255;
+    }
+    // Set system callbacks
+    ViewPort* view_port = view_port_alloc();
+    view_port_draw_callback_set(view_port, clock_render_callback, &state_mutex);
+    view_port_input_callback_set(view_port, clock_input_callback, event_queue);
+    FuriTimer* timer = furi_timer_alloc(clock_tick, FuriTimerTypePeriodic, event_queue);
+    furi_timer_start(timer, furi_kernel_get_tick_frequency());
+    // Open GUI and register view_port
+    Gui* gui = furi_record_open("gui");
+    gui_add_view_port(gui, view_port, GuiLayerFullscreen);
+    // Main loop
+    PluginEvent event;
+    for(bool processing = true; processing;) {
+        FuriStatus event_status = furi_message_queue_get(event_queue, &event, 100);
+        ClockState* plugin_state = (ClockState*)acquire_mutex_block(&state_mutex);
+        if(event_status == FuriStatusOk) {
+            // press events
+            if(event.type == EventTypeKey) {
+                if(event.input.type == InputTypeShort || event.input.type == InputTypeRepeat) {
+                    switch(event.input.key) {
+                    case InputKeyUp:
+                        if(timerStarted) timerSecs = timerSecs + 5;
```

Comment:
Why should you change the running timer?

Reply:
In case it was started late or early

Reply:
To me this behavior is not intuitive and its not documented/hinted in the gui either. I would remove it.

Reply:
Removed